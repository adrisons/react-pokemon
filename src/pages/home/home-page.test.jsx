import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import HomePage from "./home-page";

describe("GIVEN: HomePage", () => {
  const component = renderer.create(
    <Router>
      <HomePage />
    </Router>
  );
  const tree = component.toJSON();
  it("THEN: should match snapshot", () => {
    expect(tree).toMatchSnapshot();
  });

  describe("WHEN: data not loaded", () => {
    beforeEach(() => {
      jest.spyOn(window, "fetch").getMockImplementation(
        () =>
          new Promise((resolve, reject) => {
            resolve();
          })
      );
    });
    it("THEN: should display loading", () => {
      expect(tree.props.className).toEqual("loading");
    });
  });
});
