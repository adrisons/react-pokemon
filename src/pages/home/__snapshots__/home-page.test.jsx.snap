// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIVEN: HomePage THEN: should match snapshot 1`] = `
<div
  className="loading"
>
  <img
    alt="loading"
    src="pokeball.png"
  />
</div>
`;
